### DISCUSSION TITLE
Is there a way for every proposal to have its own unique list of whitelisted voters?

### DISCUSSION THREAD

**Discussion by rho-cassiopeiae** [snapshot-labs/snapshot#3272] [Q&A] [answered]
As far as I can understand, the whitelist strategy is static – you specify who you allow to vote _once_ and the list stays the same for every proposal.

What if I want to randomly choose, say, 10 voters for every proposal using VRF and count only their votes on the proposal and no one else's? Is this possible?

I think it would be possible if the strategy function's input included proposal id or something. Then you would be able to make an api/subgraph/contract call to get the data necessary to assign appropriate weights, but as far I know there's no way to pass some additional input to the strategy.

**Comment by samwhosung** [comment, marked as answer]
@rho-cassiopeiae you could change the whitelist before you create each proposal. The settings will be saved on proposal level each time you create one so you can change them after creating each proposal.

I think what you are trying to do would be a perfect candidate for the upcoming feature we call "Vote validation" these will work in combination with strategies and allow much more advanced settings in terms of who can vote.
